rcl_lifecycle/include/rcl_lifecycle/data_types.h

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL_LIFECYCLE__DATA_TYPES_H_
+#define RCL_LIFECYCLE__DATA_TYPES_H_
+
+#include <rcl/rcl.h>
+#include "rcl_lifecycle/visibility_control.h"
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @brief simple definition of a state
+ * @param state: integer giving the state
+ * @param label: label for easy indexing
+ */
+typedef struct _rcl_lifecycle_state_t
+{
+  const char * label;
+  unsigned int id;
+} rcl_lifecycle_state_t;
+
+/**
+ * @brief transition definition
+ * @param start: rcl_lifecycle_state_t as a start state
+ * @param goal: rcl_lifecycle_state_t as a goal state
```

Comment:
I'm not sure you should use the `@param` directive to describe members of the struct. There is a member documentation syntax for doxygen, see http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html (search for `Putting documentation after members` as there doesn't seem to be an achor).